src/transport/SessionManager.h

```diff
@@ -217,9 +219,9 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate
     }
 
     // TODO: implements group sessions
-    Optional<SessionHandle> CreateGroupSession(GroupId group, chip::FabricIndex fabricIndex)
+    Optional<SessionHandle> CreateGroupSession(GroupId group, chip::FabricIndex fabricIndex, NodeId sourceNodeId)
```

Comment:
Why are we requring consumers to pass in the node id, if it can be recoveted from the fabricIndex?

Reply:
Because, of how you get access to the fabric table. You unfortunately cannot access it without calling the Server instance (which might not be present if you're a controller or commissionner ) and thus creating a dependency between the App layer and the Transport layer. Should we have a global variable for the fabric table instead of a specific instance hidden somewhere in the app layer, all of this non-sense of passing the sourceNodeId could go away and a direct call could be added to the SessionManager like this `FabricTable.FindFabricWithIndex(fabricIndex)->GetPeerId()`

As for this specific solution to the problem stated aboved, it was either that (passing sourceNodeId) or adding a fabricTable pointer to the constructor of SessionManager, which I've already modified in order to add a pointer for the persistent storage delegate.